source.go

```diff
@@ -12,87 +12,71 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package source
+//go:generate paramgen -output=paramgen_src.go SourceConfig
+
+package postgres
 
 import (
 	"context"
 	"fmt"
 
+	"github.com/conduitio/conduit-connector-postgres/source"
 	"github.com/conduitio/conduit-connector-postgres/source/logrepl"
 	"github.com/conduitio/conduit-connector-postgres/source/longpoll"
 	sdk "github.com/conduitio/conduit-connector-sdk"
 	"github.com/jackc/pgx/v4"
 )
 
-var (
-	_ Iterator = (*logrepl.CDCIterator)(nil)
-	_ Iterator = (*longpoll.SnapshotIterator)(nil)
-)
-
 // Source is a Postgres source plugin.
 type Source struct {
 	sdk.UnimplementedSource
 
-	iterator Iterator
-	config   Config
+	iterator source.Iterator
+	config   SourceConfig
 	conn     *pgx.Conn
 }
 
+type SourceConfig struct {
+	// URL is the connection string for the Postgres database.
+	URL string `json:"url" validate:"required"`
+	// The name of the table in Postgres that the connector should read.
+	Table string `json:"table" validate:"required"`
+	// Comma separated list of column names that should be included in each Record's payload.
+	Columns []string `json:"columns"`
+	// Column name that records should use for their `Key` fields.
+	Key string `json:"key"`
+
+	// Whether or not the plugin will take a snapshot of the entire table before starting cdc mode.
+	SnapshotMode source.SnapshotMode `json:"snapshotMode" validate:"inclusion=initial|never" default:"initial"`
```

Comment:
I know this is out of the scope and a bit too late to change, but can't understand why we didn't just go with a boolean here 😅

Reply:
I think the idea was to give us the freedom to add more modes in the future, e.g. [debezium has 6](https://debezium.io/documentation/reference/stable/connectors/postgresql.html#postgresql-connector-snapshot-mode-options).